examples/vision/mlp_image_classification.py

```diff
@@ -0,0 +1,452 @@
+You may also try to increase the size of the input images and use different patch sizes.
+"""
+
+"""
+## The FNet model
+
+The FNet uses a similar block to the Transformer block. However, FNet replaces the self-attention layer
+in the Transformer block with a parameter-free 2D Fourier transformation layer:
+
+1. one 1D Fourier Transform is applied along the patches.
+2. one 1D Fourier Transform is applied along the channels.
+"""
+
+"""
+### Implement the FNet module
+"""
+
+
+class FNetLayer(layers.Layer):
+    def __init__(self, num_patches, embedding_dim, dropout_rate, *args, **kwargs):
+        super(FNetLayer, self).__init__(*args, **kwargs)
+
+        self.fourier = layers.Lambda(
+            lambda inputs: tf.cast(
+                tf.signal.rfft2d(
+                    inputs, fft_length=[num_patches, (embedding_dim * 2) - 1]
+                ),
+                dtype=tf.dtypes.float32,
+            )
+        )
+
+        self.ffn = keras.Sequential(
+            [
+                layers.Dense(units=embedding_dim),
+                tfa.layers.GELU(),
+                layers.Dropout(rate=dropout_rate),
+            ]
+        )
+
+        self.normalize1 = layers.LayerNormalization(epsilon=1e-6)
+        self.normalize2 = layers.LayerNormalization(epsilon=1e-6)
+
+    def call(self, inputs):
+        # Apply fourier transformations.
+        x = self.fourier(inputs)
+        # Add skip connection.
+        x = x + inputs
+        # Apply layer normalization.
+        x = self.normalize1(x)
+        # Apply Feedfowrad network.
+        x_ffn = self.ffn(x)
+        # Add skip connection.
+        x = x + x_ffn
+        # Apply layer normalization.
+        return self.normalize2(x)
+
+
+"""
+### Build, train, and evaluate a FNet model
+
+Note that training the model with the current settings on a V100 GPUs
+takes around 25 seconds per epoch.
+"""
+
+fnet_blocks = keras.Sequential(
+    [FNetLayer(num_patches, embedding_dim, dropout_rate) for _ in range(num_blocks)]
+)
+
+fnet_classifier = build_classifier(fnet_blocks, positional_encoding=True)
+history = run_experiment(fnet_classifier)
+
+"""
+The FNet model achieves around 45% accuracy and 77% top-5 accuracy on the test data.
+
+As shown in the [FNet](https://arxiv.org/abs/2105.03824) paper,
+better results can be achieved by increasing the embedding dimensions,
+increasing, increasing the number of FNet blocks, and training the model for longer.
+You may also try to increase the size of the input images and use different patch sizes.
+The FNet scales very efficiently to long inputs, runs much faster faster than attention-based
+Transformer models, and produces competitive accuracy results.
+"""
+
+"""
+## The gMLP model
+
+The gMLP as an MLP architecture with Spatial Gating Unit (SGU).
+The SGU  enables cross-patch interactions across the spatial (channel) dimension by:
+1. Transform the input spatially by applying linear projection across patches (along channels).
```

Comment:
Add line break abov (before start of list)

Reply:
Done